File: backend/src/main/kotlin/ch/streckeisen/mycv/backend/coverletter/CoverLetterGenerationRequestDto.kt
package ch.streckeisen.mycv.backend.coverletter

data class CoverLetterGenerationRequestDto(
    val language: String?,
    val style: String?,
    val mirrorProfileImage: Boolean?,
    val application: CoverLetterApplicationDto?,
    val attachedDocuments: List<String?>?
)

File: frontend/src/types/coverletter/CoverLetterGenerationRequestDto.ts
import { CoverLetterApplicationDto } from '@/types/coverletter/CoverLetterApplicationDto.ts'

export interface CoverLetterGenerationRequestDto {
  language: string | undefined
  style: string | undefined
  mirrorProfileImage: boolean | undefined
  application: CoverLetterApplicationDto | undefined
  documents: string[] | undefined
}